src/main/java/org/mule/service/scheduler/internal/threads/UberSchedulerThreadPools.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.service.scheduler.internal.threads;
+
+import static java.lang.String.format;
+import static java.lang.System.lineSeparator;
+import static java.lang.Thread.currentThread;
+import static java.util.Arrays.asList;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.mule.service.scheduler.ThreadType.IO;
+
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.scheduler.Scheduler;
+import org.mule.runtime.api.scheduler.SchedulerConfig;
+import org.mule.runtime.api.scheduler.SchedulerPoolsConfig;
+import org.mule.service.scheduler.internal.DefaultScheduler;
+import org.mule.service.scheduler.internal.ThrottledScheduler;
+import org.mule.service.scheduler.internal.executor.ByCallerThreadGroupPolicy;
+import org.mule.service.scheduler.internal.executor.ByCallerThrottlingPolicy;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+
+/**
+ * {@link SchedulerThreadPools} implementation that uses one single pool for cpuLight, IO and cpuIntensive pools, while
```

Comment:
single pool for cpuLight, IO and cpuIntensive pools -> single pool for cpuLight, IO and cpuIntensive tasks